modules/ghlint-snakeyaml/src/test/kotlin/net/twisterrob/ghlint/model/SnakePermissionsTest.kt

```diff
@@ -0,0 +1,151 @@
+package net.twisterrob.ghlint.model
+
+import io.kotest.matchers.shouldBe
+import io.kotest.matchers.shouldNotBe
+import net.twisterrob.ghlint.testing.load
+import net.twisterrob.ghlint.testing.loadUnsafe
+import org.junit.jupiter.api.Test
+
+class SnakePermissionsTest {
+
+	private fun File.asJob(): Job.NormalJob = (content as Workflow).jobs.values.single() as Job.NormalJob
+
+	@Test fun `job has permissions with correct values`() {
+		val job = load(
+			"""
+				on: push
+				jobs:
+				  test:
+				    permissions:
+				      contents: read
+				      issues: write
+				    runs-on: ubuntu-latest
+				    steps:
+				      - uses: actions/checkout@v4
+			""".trimIndent()
+		).asJob()
+
+		job.permissions?.contents shouldBe Access.READ
+		job.permissions?.issues shouldBe Access.WRITE
+	}
+
+	@Test fun `workflow has permissions with correct values`() {
+		val workflow = load(
+			"""
+				on: push
+				permissions:
+				  pull-requests: write
+				  id-token: read
+				jobs:
+				  test:
+				    runs-on: ubuntu-latest
+				    steps:
+				      - uses: actions/checkout@v4
+			""".trimIndent()
+		).content as Workflow
+
+		workflow.permissions?.pullRequests shouldBe Access.WRITE
+		workflow.permissions?.idToken shouldBe Access.READ
+	}
```

Comment:
add test for each `val get()`

it's a lot of tests, so I think just assertions with random Access values is good enough. Use `: none` as well.